### PR TITLE
Do not re-download iso multiple times from different urls

### PR DESCRIPTION
**Pull request by VladRassokhin**
In case of two or more iso_urls check for downloaded files prior to downloading them.
Speedups case when some iso already downloaded and another url prepended to iso_urls list.

Tested that locally, it works on my machine ;) 
Though logic is quite simple.
Any ideas how to write test for that?

### CODE REVIEW

**Comment by VladRassokhin** [comment]
Comments and suggestions how to improve change are welcomed

File: common/step_download.go [hunk 1]
```diff
@@ -90,19 +92,34 @@ func (s *StepDownload) Run(state multistep.StateBag) multistep.StepAction {
...
+			ui.Message(fmt.Sprintf("Found already downloaded, initial checksum matched, no download needed: %s", url))
+			finalPath = config.TargetPath
+			break
 		}
+	}
 
-		if !retry {
-			return multistep.ActionHalt
-		}
+	if finalPath == "" {
```
Comment (SwampDragons) [inline common/step_download.go]: Why add a second loop here and introduce the need to track the download configs?  Why not just do the check and then the download inside the same loop?  This feels like unnecessary complexity on a first read.

Comment (VladRassokhin) [reply common/step_download.go]: The whole PR is about adding this second loop. Maybe I've poorly described problem. 
So imagine next case:
1. There're two urls in 'iso_urls', first one accessible only in internal network (link to local cache actually), second is accessible from anywhere.
2. Running packer build from place where first link doesn't work, got file downloaded from second one and saved as `packer_cache/some_hash_of_second_url.iso`
3. Then running packer in place where first link is accessible, got packer re-downloaded file and stored it as `packer_cache/some_hash_of_first_url.iso`. But obviously packer could have use `packer_cache/some_hash_of_second_url.iso`.

So my PR is to change behaviour on third step, to check for ALL downloaded files prior to ANY download.

Comment (SwampDragons) [reply common/step_download.go]: Ah you're right I read this too fast and misunderstood the goal.

**Comment by VladRassokhin** [comment]
Ok, thanks for merging.

**Pull request merged**